### PR TITLE
tests: drivers: spi: spi_slave: Extend test scope

### PR DESCRIPTION
**Pull request by nordic-segl**
Add possibility to test all four SPI modes (CPOL, CPHA).
Add possibility to test different bitrates.
Add tests that checks handling of zero length buffers on master.

### CODE REVIEW

**Review by nordic-piks — APPROVED** [review]

**Review by tbursztyka — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

File: tests/drivers/spi/spi_slave/boards/nrf54h20dk_nrf54h20_common.dtsi [hunk 1]
```diff
@@ -60,10 +60,10 @@ dut_spi: &spi130 {
 	pinctrl-names = "default", "sleep";
 	overrun-character = <0x00>;
 	cs-gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+	wake-gpios = <&gpio0 2 (GPIO_OPEN_SOURCE | GPIO_ACTIVE_HIGH)>;
```
Comment (anangl) [inline tests/drivers/spi/spi_slave/boards/nrf54h20dk_nrf54h20_common.dtsi:63]: `GPIO_OPEN_SOURCE` is unneeded here and actually a bit misleading (the pin is not configured this way). This flag will be ignored by the driver anyway.
But for the wake mechanism to work, `wake-gpios` property needs to be configured on both sides. If this actually works in the current form, it must be by chance.

Comment (nordic-segl) [reply tests/drivers/spi/spi_slave/boards/nrf54h20dk_nrf54h20_common.dtsi:63]: I will remove GPIO_OPEN_SOURCE from DTS.

I failed to get wakeup gpio working on both sides. I created task NRFX-5836 to prepare sample demonstrating how to use this feature.
In the current form, master prints warning that timeout occurred while waiting for confirmation from slave. However, master ignores timeout and sends data.
When I also set wakeup gpio on slave, test hangs. Slave doesn't see wakeup signal from master and hangs in an infinite loop. I lack skills to investigate this issue.
It may be due to invalid GPIO configuration / missing GPIOTE configuration / synchronization issue (i.e. force master to wait longer for confirmation from slave) / etc.

**Review by nordic-krch — DISMISSED** [review]

File: tests/drivers/spi/spi_slave/boards/nrf54h20dk_nrf54h20_common.dtsi (hunk 1 above)
Comment (anangl) [reply tests/drivers/spi/spi_slave/boards/nrf54h20dk_nrf54h20_common.dtsi]: > In the current form, master prints warning that timeout occurred while waiting for confirmation from slave. However, master ignores timeout and sends data.

What's the point in adding this property then?

Comment (nordic-segl) [reply tests/drivers/spi/spi_slave/boards/nrf54h20dk_nrf54h20_common.dtsi]: I removed wake-gpios as it has no impact on test result.

**Review by anangl — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

**Comment by nordic-segl** [comment]
@tbursztyka Could You take a look?

**Review by tbursztyka — DISMISSED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
This should be using inclusive language for naming the test (spi_peripheral). Please see https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language

**Comment by nordic-segl** [comment]
Test was renamed to spi_controller_peripheral as there are two instances connected together.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@nordic-segl can you please rebase this?

**Comment by nordic-segl** [comment]
Test was renamed to spi_controller_peripheral as there are two instances connected together.

> @nordic-segl can you please rebase this?

Done.

**Comment by nordic-segl** [comment]
tested locally on nrf54l15pdk - known issue with 8MHz (to be investigated later) 5 remaining test configurations are passing.

./zephyr/scripts/twister --ninja --inline-logs --jobs 1 -T zephyr/tests/drivers/spi/spi_controller_peripheral/ --platform nrf54l15pdk/nrf54l15/cpuapp --device-testing --device-serial /dev/serial/by-id/usb-SEGGER_J-Link_001057750650-if02 --no-clean -v -v --fixture gpio_spi_loopback

**Comment by nordic-segl** [comment]
tested locally on nrf54h20dk - all test configurations are passing.

./zephyr/scripts/twister --ninja --inline-logs --jobs 1 -T zephyr/tests/drivers/spi/spi_controller_peripheral/ --platform nrf54h20dk/nrf54h20/cpuapp --device-testing --device-serial /dev/serial/by-id/usb-SEGGER_J-Link_001051114155-if00 --no-clean -v -v --fixture gpio_spi_loopback
